internal/acceptance_test/resource_host_test.go

```diff
@@ -53,17 +66,92 @@ resource "hpegl_metal_host" "test_host" {
   machine_size       = "${data.hpegl_metal_available_resources.compute.machine_sizes.0.name}"
   ssh                = ["User1 - Linux"]
   networks           = sort(local.networks)
+  network_route      = "Public"
   location           = var.location
   description        = "hello from Terraform"
 }
 `
 }
 
-func testWaitUntilHostCreated() resource.TestCheckFunc {
+// testAccHostUpdateConfig updates the terraform config by updating
+// attributes of the host 'test_host':
+//   - description is updated
+//   - last network removed from sorted list of networks
+// Updated config is compared against config specified in testAccCheckHostBasic.
+func testAccHostUpdateConfig() string {
+	return `
+provider "hpegl" {
+	metal {
+	}
+}
+
+variable "location" {
+	default = "USA:Central:V2DCC01"
+}
+
+data "hpegl_metal_available_resources" "compute" {
+}
+
+locals  {
+	host_os_flavor = "${data.hpegl_metal_available_resources.compute.images.0.flavor}"
+	host_os_version = "${data.hpegl_metal_available_resources.compute.images.0.version}"
+	networks = ([for net in "${data.hpegl_metal_available_resources.compute.networks}": 
+		net.name if net.location == var.location] )
+	sorted_networks = sort(local.networks)
+	updated_networks_length = length(local.sorted_networks) - 1
+	updated_networks = ([for i, net in local.sorted_networks : net if i < local.updated_networks_length])
+}
+
+resource "hpegl_metal_host" "test_host" {
+	name               = "test"
```

Comment:
since most of this terraform block is the same as 'create', it can be modularized into something like this.

```
func hostConfig(op string) string {
	common := `
provider "hpegl" {
	metal {
	}
}
variable "location" {
	default = "USA:Central:V2DCC01"
}
data "hpegl_metal_available_resources" "compute" {
}
locals  {
	host_os_flavor = "${data.hpegl_metal_available_resources.compute.images.0.flavor}"
	host_os_version = "${data.hpegl_metal_available_resources.compute.images.0.version}"
	networks = ([for net in "${data.hpegl_metal_available_resources.compute.networks}": 
		net.name if net.location == var.location] )
	sorted_networks = sort(local.networks)
	updated_networks_length = length(local.sorted_networks) - 1
	updated_networks = ([for i, net in local.sorted_networks : net if i < local.updated_networks_length])	
}`

	description := "hello from Terraform"
	if op == "update" {
		description = "hello from Terraform(updated)"
	}

	networks := `sort(local.networks)`
	if op == "update" {
		networks = `local.updated_networks`
	}

	host := fmt.Sprintf(`
resource "hpegl_metal_host" "test_host" {
	name               = "test"
	image              = join("@",[local.host_os_flavor, local.host_os_version])
	machine_size       = "${data.hpegl_metal_available_resources.compute.machine_sizes.0.name}"
	ssh                = ["User1 - Linux"]
	networks           = %s
	network_route      = "Public"
	location           = var.location
	description        = %s
}`, networks, description)

	return (common + host)
}
```